tools/gen_ort_dockerfile.py

```diff
@@ -266,48 +267,21 @@ def dockerfile_for_linux(output_file):
     if FLAGS.ort_openvino is not None:
         df += '''
 # OpenVino specific headers and libraries
-RUN cp -r /opt/intel/openvino_${ONNXRUNTIME_OPENVINO_VERSION}/licensing \
-       /opt/onnxruntime/LICENSE.openvino
+RUN cp -r /opt/intel/openvino_2022/docs/licensing /opt/onnxruntime/LICENSE.openvino
 
 RUN cp /workspace/onnxruntime/include/onnxruntime/core/providers/openvino/openvino_provider_factory.h \
        /opt/onnxruntime/include
 
-# libonnx_proto.so, libprotobuf.so.3.7.1.0 are needed when openvino execution provider is used
-RUN if [ -f /opt/intel/openvino_${ONNXRUNTIME_OPENVINO_VERSION}/deployment_tools/ngraph/lib/libonnx_proto.so ]; then \
-        cp /opt/intel/openvino_${ONNXRUNTIME_OPENVINO_VERSION}/deployment_tools/ngraph/lib/libonnx_proto.so \
-            /opt/onnxruntime/lib; \
-        cp /opt/intel/openvino_${ONNXRUNTIME_OPENVINO_VERSION}/deployment_tools/ngraph/lib/libprotobuf.so.3.7.1.0 \
-            /opt/onnxruntime/lib; \
-        (cd /opt/onnxruntime/lib && ln -sf libprotobuf.so.3.7.1.0 libprotobuf.so); \
-    fi
-
 RUN cp /workspace/build/${ONNXRUNTIME_BUILD_CONFIG}/libonnxruntime_providers_openvino.so \
        /opt/onnxruntime/lib && \
-    cp /opt/intel/openvino_${ONNXRUNTIME_OPENVINO_VERSION}/deployment_tools/inference_engine/lib/intel64/libinference_engine.so \
-       /opt/onnxruntime/lib && \
-    cp /opt/intel/openvino_${ONNXRUNTIME_OPENVINO_VERSION}/deployment_tools/inference_engine/lib/intel64/libinference_engine_legacy.so \
-       /opt/onnxruntime/lib && \
-    cp /opt/intel/openvino_${ONNXRUNTIME_OPENVINO_VERSION}/deployment_tools/inference_engine/lib/intel64/libinference_engine_transformations.so \
-       /opt/onnxruntime/lib && \
-    cp /opt/intel/openvino_${ONNXRUNTIME_OPENVINO_VERSION}/deployment_tools/ngraph/lib/libngraph.so \
-       /opt/onnxruntime/lib && \
-    cp /opt/intel/openvino_${ONNXRUNTIME_OPENVINO_VERSION}/deployment_tools/ngraph/lib/libonnx_importer.so \
-       /opt/onnxruntime/lib && \
-    cp /opt/intel/openvino_${ONNXRUNTIME_OPENVINO_VERSION}/deployment_tools/inference_engine/external/tbb/lib/libtbb.so.2 \
-       /opt/onnxruntime/lib && \
-    cp /opt/intel/openvino_${ONNXRUNTIME_OPENVINO_VERSION}/deployment_tools/inference_engine/lib/intel64/plugins.xml \
-       /opt/onnxruntime/lib && \
-    cp /opt/intel/openvino_${ONNXRUNTIME_OPENVINO_VERSION}/deployment_tools/inference_engine/lib/intel64/libMKLDNNPlugin.so \
-       /opt/onnxruntime/lib && \
-    cp /opt/intel/openvino_${ONNXRUNTIME_OPENVINO_VERSION}/deployment_tools/inference_engine/lib/intel64/libinference_engine_lp_transformations.so \
-       /opt/onnxruntime/lib && \
-    cp /opt/intel/openvino_${ONNXRUNTIME_OPENVINO_VERSION}/deployment_tools/inference_engine/lib/intel64/libinference_engine_ir_reader.so \
+    cp /opt/intel/openvino_2022/runtime/lib/intel64/lib*.so \
```

Comment:
Why we are copying all the libraries here? Can we only copy the subset that might be sufficient like we used to do before?
There might be some runtime libraries designed for different hardware that we don't support. I am concerned about the size and possible symbol collisions it may cause. If there aren't many unneeded libraries, I think this way of copying all the libraries future-proofs us.

Reply:
Yes. The list is trimmed down.